### PR TITLE
[EMB-160] Home page

### PR DESCRIPTION
**Pull request by binoculars**
## TODO
- [x] Goodbye route
- [x] Use static background image used ~~Background canvas moving lines and dots (or remove it)~~
- [x] Integrate ember-cp-validations
- [x] YouTube modal
- [x] Refine styles
- [x] Add ember-cli-build option to unbundle Zxcvbn

## Purpose

Convert the home/landing page to ember

## Summary of Changes

Makes the dashboard page its own route, `/dashboard`. This mostly necessary due to the way ember-routing works. The behavior is preserved. If the user is logged out and goes to the `/dashboard` route, they'll be transitioned to the `/` route, and vice-versa if they are logged in.

Fixes the jank on the OSF logo when mousing over it. This is due to manipulating the `top` and `left` CSS properties, which cause a repaint. This is now done with the `transform: translate()` property.

#### [ember-cli-password-strength](/elwayman02/ember-cli-password-strength) [Ember Observer Score](https://emberobserver.com/addons/ember-cli-password-strength)
Uses the Zxcvbn library from Dropbox. We currently use this on the OSF to determine password strength.

#### [ember-g-recaptcha](/algonauti/ember-g-recaptcha) [Ember Observer Score](https://emberobserver.com/addons/ember-g-recaptcha)
Needed to handle reCaptcha

#### [ember-bootstrap-datepicker](/topaxi/ember-bootstrap-datepicker)  [Ember Observer Score](https://emberobserver.com/addons/ember-bootstrap-datepicker)
Needed for validated-input component from ember-osf

#### [ember-cp-validations](/offirgolan/ember-cp-validations) [Ember Observer Score](https://emberobserver.com/addons/ember-cp-validations)
Input validation

#### [ember-i18n-cp-validations](/jasonmit/ember-i18n-cp-validations) [Ember Observer Score](https://emberobserver.com/addons/ember-i18n-cp-validations)
Internationalization for input validation

#### [ember-component-attributes](/mmun/ember-component-attributes) [Ember Observer Score](https://emberobserver.com/addons/ember-component-attributes)
Needed to add `aria-label` to input components
h/t @brianjgeiger 

#### [ember-youtube](/oskarrough/ember-youtube) [Ember Observer Score](https://emberobserver.com/addons/ember-youtube)
Needed for the YouTube player in the modal on the home page

## Ticket

https://openscience.atlassian.net/browse/EMB-160

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/locales/en/translations.ts [hunk 1]
```diff
@@ -285,4 +286,137 @@ export default {
...
+        integrations_alt_mendeley: 'Mendeley logo',
+        features_title: '<strong>Everything</strong> your research needs to be a success',
+        features_manage_title: 'Manage your project',
+        features_manage_paragraph: 'View all of your projects from <strong>one dashboard.</strong>',
+        features_share_title: 'Quickly share files',
+        features_share_paragraph: '<strong>Share key project information</strong> and allow others to use and cite it.',
+        features_changes_title: 'See project changes',
+        features_changes_paragraph: 'See the latest project changes, who is contributing and <strong>historical file versions.</strong>',
+        features_analytics_title: 'View project analytics',
+        features_analytics_paragraph: 'Access<strong>project data</strong> ranging from visits over time to top referring websites.',
```
Comment (aaxelb) [inline app/locales/en/translations.ts]: Add a space after 'Access'

File: app/components/validated-input/component.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    Text = 'text',
+    TextArea = 'textarea',
+    ReCaptcha = 'reCaptcha',
+}
+
+export default class ValidatedInput extends Component.extend({
+    classNames: ['validated-input'],
+    classNameBindings: ['showErrorClass:has-error', 'isValid:has-success'],
+
+    disabled: false,
```
Comment (aaxelb) [inline app/components/validated-input/component.ts]: It'd be nice to move these initializations to the class body, where they're defined/typed. The lazy way I've done it before is:
```ts
disabled: boolean = this.disabled || false;
```
but that's brittle for falsy values. I've seen some people use lodash:
```ts
disabled: boolean = _.defaultTo(this.disabled, false);
```
but that's yet another dependency. Could make our own `defaultTo` util, I guess, but that's still an extra import everywhere... Not sure, but that might be more appealing to me than these duplicate lists of properties.

File: app/components/sign-up-policy/styles.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+a {
+    color: #5bc0de;
```
Comment (jamescdavis) [inline app/components/sign-up-policy/styles.scss]: use variable instead of color literal

File: app/components/validated-input/component.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+enum InputType {
+    Date = 'date',
+    Password = 'password',
+    Text = 'text',
+    TextArea = 'textarea',
+    ReCaptcha = 'reCaptcha',
+}
+
+export default class ValidatedInput extends Component.extend({
+    classNames: ['validated-input'],
```
Comment (jamescdavis) [inline app/components/validated-input/component.ts]: use classnames decorator

File: app/components/validated-input/component.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    classNameBindings: ['showErrorClass:has-error', 'isValid:has-success'],
```
Comment (jamescdavis) [inline app/components/validated-input/component.ts]: use classname decorator

File: app/components/validated-input/style.scss [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// stylelint-disable selector-class-pattern
+.datepicker.dropdown-menu {
+    background-color: #fff;
```
Comment (jamescdavis) [inline app/components/validated-input/style.scss]: use variable

File: app/dashboard/route.ts [hunk 7]
```diff
@@ -1,3 +1,14 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
-export default class Dashboard extends Route {}
+export default class Dashboard extends Route.extend({
+    session: service('session'),
```
Comment (jamescdavis) [inline app/dashboard/route.ts]: service decorator

**Comment by jamescdavis** [comment]
Passing build: https://travis-ci.org/binoculars/ember-osf-web/builds/356381833

**Pull request merged**